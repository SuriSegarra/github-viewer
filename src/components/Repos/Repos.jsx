import React from 'react';
import PropTypes from 'prop-types';
import Repo from '../repo/Repo';

const Repos = ({ repos }) => {
  const repoData = repos.map(repo => <Repo key={repo.id} {...repo} />);
  return (
    <ul>
      {repoData}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    html_url:PropTypes.string.isRequired
  })).isRequired
};

export default Repos;

