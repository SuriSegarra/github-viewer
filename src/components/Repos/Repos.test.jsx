import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos', () => {
  it('matches repos snapshot', () => {
    const repos = [
      {
        id:1,
        name: '-recipe-app',
        html_url:'https://github.com/SuriSegarra/-recipe-app'
      },
      {
        id:2,
        name: 'About-Me',
        html_url:'https://github.com/SuriSegarra/About-Me'
      }
    ];
    const wrapper = shallow(<Repos repos={repos}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
