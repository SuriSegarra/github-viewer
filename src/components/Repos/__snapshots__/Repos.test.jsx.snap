// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos matches repos snapshot 1`] = `
<ul>
  <Repo
    html_url="https://github.com/SuriSegarra/-recipe-app"
    id={1}
    key="1"
    name="-recipe-app"
  />
  <Repo
    html_url="https://github.com/SuriSegarra/About-Me"
    id={2}
    key="2"
    name="About-Me"
  />
</ul>
`;
